test/e2e/commatrix_suite_test.go

```diff
@@ -0,0 +1,162 @@
+package e2e
+
+import (
+	"context"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"golang.org/x/sync/errgroup"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/openshift-kni/commatrix/pkg/client"
+	commatrixcreator "github.com/openshift-kni/commatrix/pkg/commatrix-creator"
+	"github.com/openshift-kni/commatrix/pkg/consts"
+	"github.com/openshift-kni/commatrix/pkg/endpointslices"
+	"github.com/openshift-kni/commatrix/pkg/types"
+	"github.com/openshift-kni/commatrix/pkg/utils"
+	"github.com/openshift-kni/commatrix/test/pkg/firewall"
+	node "github.com/openshift-kni/commatrix/test/pkg/node"
+)
+
+var (
+	cs           *client.ClientSet
+	matrix       *types.ComMatrix
+	isSNO        bool
+	utilsHelpers utils.UtilsInterface
+	nodeList     *corev1.NodeList
+	artifactsDir string
+)
+
+const (
+	workerNodeRole = "worker"
+	tableName      = "table inet openshift_filter"
+	chainName      = "chain OPENSHIFT"
+)
+
+var _ = BeforeSuite(func() {
+	By("create output folder")
+	artifactsDir = os.Getenv("ARTIFACT_DIR")
+	if artifactsDir == "" {
+		log.Println("env var ARTIFACT_DIR is not set, using default value")
+	}
+	artifactsDir = filepath.Join(artifactsDir, "commatrix")
+
+	err := os.MkdirAll(artifactsDir, 0755)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("generating the commatrix")
+	cs, err = client.New()
+	Expect(err).NotTo(HaveOccurred())
+
+	utilsHelpers = utils.New(cs)
+
+	deployment := types.Standard
+	isSNO, err = utilsHelpers.IsSNOCluster()
+	Expect(err).NotTo(HaveOccurred())
+
+	if isSNO {
+		deployment = types.SNO
+	}
+
+	infra := types.Cloud
+	isBM, err := utilsHelpers.IsBMInfra()
+	Expect(err).NotTo(HaveOccurred())
+
+	if isBM {
+		infra = types.Baremetal
+	}
+
+	epExporter, err := endpointslices.New(cs)
+	Expect(err).ToNot(HaveOccurred())
+
+	By("Creating commMatrix")
+	commMatrix, err := commatrixcreator.New(epExporter, "", "", infra, deployment)
+	Expect(err).NotTo(HaveOccurred())
+
+	matrix, err = commMatrix.CreateEndpointMatrix()
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Creating Namespace")
+	err = utilsHelpers.CreateNamespace(consts.DefaultDebugNamespace)
+	Expect(err).ToNot(HaveOccurred())
+
+	nodeList = &corev1.NodeList{}
+	err = cs.List(context.TODO(), nodeList)
+	Expect(err).ToNot(HaveOccurred())
+})
+
+var _ = AfterSuite(func() {
+	By("Deleting Namespace")
+	err := utilsHelpers.DeleteNamespace(consts.DefaultDebugNamespace)
+	Expect(err).ToNot(HaveOccurred())
+})
+
+var _ = Describe("commatrix", func() {
+	It("should apply firewall by blocking all ports except the ones OCP is listening on", func() {
```

Comment:
`ApplyRulesToNode`, `SoftRebootNodeAndWaitForDisconnect` and `RulesList` funcs are all creating pods and waiting for them to be ready, it's time consuming, please consider reusing the pods